terraform/modules/kubernetes/kubernetes.tf

```diff
@@ -30,6 +43,78 @@ resource "kubernetes_namespace" "namespace" {
   }
 }
 
+# Workload identity[1] lets us map GCP service accounts to Kubernetes service
+# accounts. We need this so that pods can use GCP API, but also AWS APIs like S3
+# via Web Identity Federation. To use the credentials, the container must fetch
+# the authentication token from the instance metadata service. Kubernetes has
+# features for automatically providing a service account token (e.g. via a
+# a mounted volume[2]), but that would be a token for the *Kubernetes* level
+# service account, and not the one we can present to AWS.
+# [1] https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity
+# [2] https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection
+
+# For each facilitator, we first create a GCP service account.
+resource "google_service_account" "execution_manager" {
+  provider = google-beta
+  # The Account ID must be unique across the whole GCP project, and not just the
+  # namespace. It must also be less than 30 characters, so we can't concatenate
+  # environment and PHA name to get something unique. Instead, we generate a
+  # random string.
+  account_id   = "prio-${random_string.account_id.result}"
+  display_name = "prio-${var.environment}-${var.peer_share_processor_name}-execution-manager"
+}
+
+resource "random_string" "account_id" {
+  length  = 16
+  upper   = false
+  number  = false
+  special = false
+}
+
+# This is the Kubernetes-level service account which we associate with the GCP
+# service account above.
+resource "kubernetes_service_account" "execution_manager" {
+  metadata {
+    name      = "${var.peer_share_processor_name}-execution-manager"
+    namespace = var.peer_share_processor_name
+    annotations = {
+      environment = var.environment
+      # This annotation is necessary for the Kubernetes-GCP service account
+      # mapping. See step 6 in
+      # https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to
+      "iam.gke.io/gcp-service-account" = google_service_account.execution_manager.email
+    }
+  }
+}
+
+# This carefully constructed string lets us refer to the Kubernetes service
+# account in GCP-level policies, below. See step 5 in
+# https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to
+locals {
+  service_account = "serviceAccount:${var.gcp_project}.svc.id.goog[${kubernetes_namespace.namespace.metadata[0].name}/${kubernetes_service_account.execution_manager.metadata[0].name}]"
```

Comment:
That's because you (hypothetically) could do this:
```
resource "kubernetes_something" "something" {
    metadata {
        name = "foo"
    }
    metadata {
        namespace = "bar"
    }
}
```
And then `metadata[0]` is the one containing `name`, `metadata[1]` is the one containing `namespace`. Even in the case where you only specify one of these blocks, the member is a list. The Kubernetes resources actually forbid setting `metadata` multiple times, but [here's](https://github.com/letsencrypt/k8s-ct/blob/eca528090e591f849c35032b5b2a7ffa5b316646/terraform/modules/routing/public/pub.tf#L37) an example of an `aws_route_table` resource that includes multiple `route` block members.